java/java.hints/src/org/netbeans/modules/java/hints/jdk/ConvertToLambdaPreconditionChecker.java

```diff
@@ -137,9 +137,48 @@ private MethodTree getMethodFromFunctionalInterface(TreePath pathToNewClassTree)
                 candidate = (MethodTree)member;
             }
         }
+        // default methods can't be implemented with a lambda
+        ExecutableElement candidateElement = (ExecutableElement) info.getTrees().getElement(new TreePath(pathToNewClassTree, candidate));
```

Comment:
So, two comments here:
 - should we rather check the method implements an abstract interface method, rather than checking it is not default? (Meaning, if the method based is not found, we don't try to do the conversion?)
 - should this be moved to be part of `ensurePreconditionsAreChecked`? (And then `passesAllPreconditions` and `passesFatalPreconditions`?)

Reply:
> should we rather check the method implements an abstract interface method, rather than checking it is not default? (Meaning, if the method based is not found, we don't try to do the conversion?)

good thinking! I focused on default methods so much that I didn't see the inverted variant. This should be more efficient in most cases abstract methods are easier to find and the hint wouldn't have to search the whole hierarchy.

> should this be moved to be part of ensurePreconditionsAreChecked? (And then passesAllPreconditions and passesFatalPreconditions?)

I haven't implemented this so far. since the method `getMethodFromFunctionalInterface` is wrong without the change - it would return a method even when the interface is not functional. The precondition checks are all working, since the existence of the method is part of the checks.

Please take another look if you agree, if not I can still make it a precondition - but we probably should also rename the method.

Reply:
@jlahoda forgot to ping